examples/accounts.py

```diff
@@ -0,0 +1,129 @@
+parser_create.add_argument('memo', help = 'explanation of transfer', nargs = argparse.REMAINDER)
+#withdraw money from account (check)
+parser_create = subparsers.add_parser('check', description = 'withdraw money via check')
+parser_create.add_argument('checkNumber', help = 'number of check')
+parser_create.add_argument('account', help = 'account from which to withdraw money')
+parser_create.add_argument('amount', help = 'amount withdrawn', type = decimal.Decimal, default = decimal.Decimal('0.00'))
+parser_create.add_argument('memo', help = 'check memo', nargs = argparse.REMAINDER)
+#deposit money into account
+parser_create = subparsers.add_parser('deposit', description = 'deposit money into account')
+parser_create.add_argument('account', help = 'account in which to deposit money')
+parser_create.add_argument('amount', help = 'total deposit', type = decimal.Decimal, default = decimal.Decimal('0.00'))
+parser_create.add_argument('memo', help = 'source of deposit', nargs = argparse.REMAINDER)
+#withdraw money from account (cash)
+parser_create = subparsers.add_parser('withdraw', description = 'withdraw amount of cash from account')
+parser_create.add_argument('account', help = 'account from which to withdraw money')
+parser_create.add_argument('amount', help = 'total withdrawl', type = decimal.Decimal, default = decimal.Decimal('0.00'))
+parser_create.add_argument('memo', help = 'reason for withdrawl', nargs = argparse.REMAINDER)
+args = parser.parse_args()
+
+#check to see if args.date is a string -- convert to datetime object if so
+if isinstance(args.date, str) == True:
+    args.date = datetime.datetime.strptime(args.date, '%Y-%m-%d')
+    
+with PersistentObjectPool(args.filename, flag='c') as pop:
+    if pop.root is None:
+        pop.root = pop.new(PersistentDict)
+    accounts = pop.root
+    if args.subcommand == 'create':
+        accounts[args.account] = [[args.date, decimal.Decimal(args.amount), 'Initial account balance']]
+        print("Created account '" + args.account + "'.")                       
+    elif args.subcommand == 'list':
+        L1 = ("Date         Amount  Balance  Memo\n"
+              "----------  -------  -------  ----")
+        print(L1)
+        accntBalance = decimal.Decimal(0)                           
+        for x in accounts[args.account]:
+            accntBalance = accntBalance + x[1]
+        for transaction in reversed(accounts[args.account]):
+            L2= "{:%Y-%m-%d}{:>9}{:>9}  {}".format(transaction[0], transaction[1], accntBalance, transaction[2])
+            print(L2)
+            accntBalance = accntBalance - transaction[1]
+    elif args.subcommand == 'transfer':
+        s= " "
+        memo = s.join(args.memo)
+        memo = memo[0].upper() + memo[1:]
+        accounts[args.pastLocation].append([args.date, -decimal.Decimal(args.transferAmount), memo])         
+        accounts[args.futureLocation].append([args.date, decimal.Decimal(args.transferAmount), memo])
```

Comment:
args.transferAmount is declared in the parser to be type decimal.Decimal.  What that means is that the parser will call decimal.Decimal on whatever the user types.  So calling decimal.Decimal on it here is redundant, it should already be a Decimal.